File: 3619-adjacent-increasing-subarrays-detection-ii/3619-adjacent-increasing-subarrays-detection-ii.cpp
class Solution {
public:
    int maxIncreasingSubarrays(vector<int>& nums) {
        int maxi=1;
        int prev=0;
        int cur=1;
        for(int i=1;i<nums.size();i++)
        {
            if(nums[i]>nums[i-1])
            {
                cur++;
            }
            else
            {
                int val=min(prev,cur);
                maxi=max(maxi,val);
                maxi=max(maxi,cur/2);
                prev=cur;
                cur=1;
            }
        }
        int val=min(prev,cur);
        maxi=max(maxi,val);
        maxi=max(maxi,cur/2);
        return maxi;
    }
};

File: 3619-adjacent-increasing-subarrays-detection-ii/3619-adjacent-increasing-subarrays-detection-ii.java
class Solution {
    public int maxIncreasingSubarrays(List<Integer> nums) {
        int prevcnt=1 , curcnt=1;
        int ans=1;
       for(int i=1;i<nums.size();i++){
            if(nums.get(i)<=nums.get(i-1)){
                int a = Math.min(prevcnt,curcnt);
                int b = Math.max(prevcnt/2,curcnt/2);
                ans = Math.max(ans,Math.max(a,b));
                prevcnt=curcnt;
                curcnt=1;
            }
            else curcnt++;
        }
        int a = Math.min(prevcnt,curcnt);
        int b = Math.max(prevcnt/2,curcnt/2);
        return ans = Math.max(ans,Math.max(a,b));
    }
}